index.js

```diff
@@ -30,6 +32,31 @@ class StoreFacility extends Base {
     return new Autobase(this.store.session(), boostrapKey, baseOpts)
   }
 
+  async clearBeeCache (bee, ckey, maxSize = 1000) {
+    let seqCheckpoint = await bee.core.getUserData(ckey)
+    seqCheckpoint = seqCheckpoint ? convFromBin(seqCheckpoint, 'number') : 0
+    const map = new LimitedMap(maxSize)
+
+    for await (const entry of bee.createHistoryStream({ gt: seqCheckpoint, lt: bee.core.length - 1 })) {
+      if (entry.type === 'del') {
+        bee.core.clear(entry.seq)
+        continue
+      }
+
+      let seq = map.get(entry.key)
+      if (!seq) {
+        const latest = await bee.get(entry.key)
+        seq = latest?.seq ?? -1
+        map.set(entry.key, seq)
+      }
+
+      if (seq === -1 || entry.seq !== seq) {
+        bee.core.clear(entry.seq)
+      }
+    }
+    await bee.core.setUserData(ckey, convIntToBin(bee.core.length - 1))
```

Comment:
if there's a failure on exit of app this will not be stored and on every execution it will crash, due to this I think before every clear call we should do `await bee.core.setUserData(ckey, convIntToBin(entry.seq))`